src/main/java/org/roaringbitmap/BitmapContainer.java

```diff
@@ -1463,37 +1463,27 @@ public ShortIterator clone() {
 
   @Override
   public boolean hasNext() {
-    return x >= 0;
+    return position >= 0;
   }
 
   @Override
   public short next() {
-    long t = w & -w;
-    short answer = (short) ((x + 1) * 64 - 1 - Long.bitCount(t - 1));
-    w ^= t;
-    while (w == 0) {
-      --x;
-      if (x < 0) {
+    int shift = Long.numberOfLeadingZeros(word) + 1;
+    short answer = (short)((position + 1) * 64 - shift);
+    word &= ~(1L << (64 - shift));

```

Comment:
As per the Java specification, I think that this is strictly equivalent to ``word &= 0xFFFFFFFFFFFFFFFEL << ( - shift);``. Or something like it. Point is, on x64 processors, this can be implemented as two instructions: a shift followed by a bitwise AND.